### PR TITLE
Constants

### PR DESCRIPTION
**Pull request by jubianchi**
~See #20~
(edited by @Hywan): Fix #20.

Just to clear out the initial PR and fix names.

I added some constants based on http://www.lemoda.net/c/maths-constants/

### CODE REVIEW

**Comment by Hywan** [comment]
Like the approach and the source. I am assigning @Metalaka as a reviewer. Let me know if it's OK for everyone.

**Comment by jubianchi** [comment]
@Hywan I just applied a rebase on the branch as it was a bit old. Everything should be ok now.

File: Context.php [hunk 1]
```diff
@@ -246,14 +246,23 @@ protected function initializeConstants()
...
+                'SQRT_PI'          => M_SQRTPI,
+                'SQRT_2'           => M_SQRT2,
+                'SQRT_3'           => M_SQRT3,
+                'LN_PI'            => M_LNPI,
+                'LOG_2E'           => M_LOG2E,
+                'LOG_10E'          => M_LOG10E,
+                'LN_2'             => M_LN2,
+                'LN_10'            => M_LN10,
+                'INVERSE_PI'       => M_1_PI,
+                'TWICE_INVERSE_PI' => M_2_PI,
```
Comment (Hywan) [inline Context.php]: Maybe 2 over PI?

Comment (jubianchi) [inline Context.php]: i like it! let me change

**Comment by jubianchi** [comment]
good for me. Need review ;)

**Comment by Hywan** [comment]
:+1: sounds good for me. Let @Metalaka do its review.

**Comment by Metalaka** [comment]
Maybe `INFINITE` can be interesting ?

Otherwise everything is good ;)

**Comment by shouze** [comment]
@Metalaka yes it's `INF` in php... so we also have `NAN` (Not a Number).

**Comment by jubianchi** [comment]
@shouze I don't think NaN would be useful in a library dedicated to math expressions.

**Comment by shouze** [comment]
@jubianchi indeed, `NAN` would only be a returned type. But, as in [matlab](http://fr.mathworks.com/help/matlab/ref/nan.html?requestedDomain=www.mathworks.com), maybe something like a `isNaN` function should be discussed?

**Comment by Metalaka** [comment]
@jubianchi are you ok with `'INFINITE' => INF`? If yes, can you update the PR?

@shouze can u open an issue to discus about `isNaN`?

**Comment by shouze** [comment]
@Metalaka yes but do we have in maths something that can not produce a number a a result of an operation? I would says yes, dividing by zero for example, but here we have to choose between raising exceptions or returning `NaN`.

In one part:
Not raising an exception & introducting some `isNaN` function allow to write more autonomous/conditionnal arithmetic formulas.

In the other part:
Raising exceptions force to write simple formulas & write more glue code to manage them.

Also, it's not excluded maybe to support the 2 kinds of behaviors if we introduce some config options?

**Comment by Hywan** [comment]
ping?

**Comment by Hywan** [comment]
What about now? Ready for a merge?

**Comment by Metalaka** [comment]
@Hywan yes, this PR is ready

**Comment by Hywan** [comment]
Good!

**Comment by Hywan** [comment]
Thanks all!

**Comment by Hywan** [comment]
@Newman101 Welcome on board, https://hoa-project.net/Community.html!

**Pull request merged**